astropy/cosmology/tests/test_units.py

```diff
@@ -217,3 +217,157 @@ def test_equivalency_context_manager():
     # check starting with only the dimensionless_redshift equivalency.
     assert len(base_registry.equivalencies) == 1
     assert str(base_registry.equivalencies[0][0]) == "redshift"
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason="Cosmology needs scipy")
+class Test_unsuitable_units:
+#Ensure that unsuitable units are rejected
+
+    def test_Om_units(self):
+        with pytest.raises(u.UnitConversionError):
+            Planck13.Om([0.,1.]*u.m)
```

Comment:
I think it may be good to at least for one instance test a scalar too - I think that will fail too, though at a different place as is (see above). But might as well be sure

Reply:
Also, please note PEP8 issues: spaces after `,` and around `*` (latter not strictly enforced but better here, I think)